src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,74 @@
+package duke;
+
+import duke.command.AddDeadlineCommand;
+import duke.command.AddEventCommand;
+import duke.command.AddTodoCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.DoneCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.exception.DukeException;
+
+/**
+ * Represents a parser that reads and parses raw command strings.
+ */
+public class Parser {
```

Comment:
Maybe it would be more meaningful if each check/parse was given a method name of its own so we can quickly find how each command is being parsed.